diesel/src/expression/sql_literal.rs

```diff
@@ -10,43 +10,130 @@ use result::QueryResult;
-pub struct SqlLiteral<ST> {
+pub struct SqlLiteral<ST, T = ()> {
     sql: String,
+    inner: T,
     _marker: PhantomData<ST>,
 }
 
-impl<ST> SqlLiteral<ST> {
+impl<ST, T> SqlLiteral<ST, T> {
     #[doc(hidden)]
-    pub fn new(sql: String) -> Self {
+    pub fn new(sql: String, inner: T) -> Self {
         SqlLiteral {
             sql: sql,
+            inner: inner,
             _marker: PhantomData,
         }
     }
+
+    /// Bind a value for use with this SQL query.
+    ///
+    /// # Safety
+    ///
+    /// This function should be used with care, as Diesel cannot validate that
+    /// the value is of the right type nor can it validate that you have passed
+    /// the correct number of parameters.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # #[macro_use] extern crate diesel;
+    /// # include!("../doctest_setup.rs");
+    /// #
+    /// # table! {
+    /// #    users {
+    /// #        id -> Integer,
+    /// #        name -> VarChar,
+    /// #    }
+    /// # }
+    /// #
+    /// # fn main() {
+    /// #     use self::users::dsl::*;
+    /// #     use diesel::dsl::sql;
+    /// #     use diesel::sql_types::{Integer, Text};
+    /// #     let connection = establish_connection();
+    /// let seans_id = users
+    ///     .select(id)
+    ///     .filter(sql("name = ").bind::<Text, _>("Sean"))
+    ///     .get_result(&connection);
+    /// assert_eq!(Ok(1), seans_id);
+    ///
+    /// let tess_id = sql::<Integer>("SELECT id FROM users WHERE name = ")
+    ///     .bind::<Text, _>("Tess")
+    ///     .get_result(&connection);
+    /// assert_eq!(Ok(2), tess_id);
+    /// # }
+    /// ```
+    ///
+    /// ### Multiple Bind Params
+    ///
+    /// ```rust
+    /// # #[macro_use] extern crate diesel;
+    /// # include!("../doctest_setup.rs");
+    ///
+    /// # table! {
+    /// #    users {
+    /// #        id -> Integer,
+    /// #        name -> VarChar,
+    /// #    }
+    /// # }
+    /// #
+    /// # fn main() {
+    /// #     use self::users::dsl::*;
+    /// #     use diesel::dsl::sql;
+    /// #     use diesel::sql_types::{Integer, Text};
+    /// #     let connection = establish_connection();
+    /// #     diesel::insert_into(users).values(name.eq("Ryan"))
+    /// #           .execute(&connection).unwrap();
+    /// let query = users
+    ///     .select(name)
+    ///     .filter(
+    ///         sql("id > ")
+    ///         .bind::<Integer,_>(1)
+    ///         .sql(" AND name <> ")
+    ///         .bind::<Text, _>("Ryan")
+    ///     )
+    ///     .get_results(&connection);
+    /// let expected = vec!["Tess".to_string()];
+    /// assert_eq!(Ok(expected), query);
+    /// # }
+    /// ```
+    pub fn bind<BindST, U>(self, bind_value: U) -> UncheckedBind<Self, U::Expression>
+    where
+        U: AsExpression<BindST>,
+    {
+        UncheckedBind::new(self, bind_value.as_expression())
+    }
+
+    pub fn sql(self, sql: &str) -> SqlLiteral<ST, Self> {
```

Comment:
I'm surprised that this isn't failing due to `#[deny(missing_docs)]`. It needs docs.

Reply:
@sgrif Do we need this method? If we don't plan on supporting.filter(sql("id > 1").sql(" AND name <> 'Ryan'"))... I think I can omit this.

Reply:
We don't *strictly speaking* need it, but I like the symmetry of having both methods on both types. It makes it more clear to me what you can call on `UncheckedBind` by seeing it here. Otherwise I have to click through to the return type to see the methods on the other type. If we just have the same methods on both, those methods (and docs) apply in both places.